src/components/atoms/BalancesCalendarEntry.tsx

```diff
@@ -0,0 +1,58 @@
+import moment from 'moment';
+import React from 'react';
+import styled from 'styled-components';
+import {IBalanceEntry} from '../../declarations/balanceEntries';
+import { currencyFormat } from '../../helpers/currency';
+
+interface IPropsCalendarEntry {
+  className?: string;
+  entry?: IBalanceEntry;
+  date: string;
+}
+
+const calendarEntry: React.FC<IPropsCalendarEntry> = (props) => {
+
+  return ((
+    <div className={props.className}>
+      <h5>{`${moment(props.date).format('D')}.`}</h5>
+      <h5>{`${props.entry && props.entry.income ? currencyFormat((props.entry.income / 100)) : ''}`}</h5>
+      <h5>{`${props.entry && props.entry.expense ? `(${currencyFormat((props.entry.expense / 100))})` : ''}`}</h5>
+  <span>{props.entry ? <hr /> : ''}</span>
+      <h5>{`${props.entry ? currencyFormat((props.entry.liquidity / 100)) : ''}`}</h5>
```

Comment:
Need to display an empty row for positioning/styling if there isn't any data to display. The first solution didnt seem to fix that, and I'm not entirely sure what you mean with the second suggestion. I propose that we take a look together later today.

Reply:
I see, had the same situation before. 

What I meant was that you could replace the line with this.

```suggestion
      <h5>{props.entry && currencyFormat((props.entry.liquidity / 100))}</h5>
```

This way, you can avoid unnecessary string interpolation, and remove stray _else_ cases in the ternary expressions.

As for the lines before where you have two conditions, simply wrap them in a paren, such as this.

```typescript
{(props.entry && props.entry.expense) && `(${currencyFormat((props.entry.expense / 100))})`}
```

Reply:
Aha, I see.
The fix worked wonders for liquidity and <hr />, but for income and expense it does not seem to treat 0 as falsy and hence renders '0' instead of an empty line: 
`<h5>{(props.entry && props.entry.income) && currencyFormat(props.entry.income / 100)}</h5>`
Any idea why?

Reply:
Ah, yes, `'0'` is truthy, even though `0` is falsy. If it is a string, you could actually use JavaScripts coercion to check if the number in the string is greater than 0. 

```javascript
'0' > 0
```

The code above actually works even though every fiber in my body screams that it shouldn't.

Reply:
I guess this is one of the reasons why "JavaScript: The Good Parts" is 176 pages long, whereas "JavaScript: The Definite Guide" is 1096 pages long :thinking:.

Reply:
Introducing another check for '0' > 0 seems to me to make the expression even uglier than the original ternary expression. I propose that we leave the income and expense tags as is, and use the `&&` for the 'hr' tag and liquidity tag. What do you think?

Reply:
Unless there is something inherently bad with using a ternary in this situation that I'm missing

Reply:
Yeah, sounds good. But you should at least be able to remove the surrounding `{\`${...}\`}` to clean it up a bit without causing any issues.

Reply:
Ending up with just `{cond ? code : ''}`

Reply:
There's nothing inherently bad with ternary expressions, I simply thought it would be cleaner to replace them with `&&`/`||` here as that way you avoid having an else clause where it is not needed.

Reply:
Great conversation! I'm learning so much. 👨‍🎓 😌

Reply:
Good catch, seems like I've slowly transitioned to Lisp midst coding :laughing:

Reply:
(Haha, (familiar (with (that (problem (!))))))